### PR TITLE
feature: add a way to always include null in the date filter

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #669 
| License       | MIT
| Doc PR        | todo

Add a way to always include null values.

### CODE REVIEW

File: src/Bridge/Doctrine/Orm/Filter/DateFilter.php [hunk 1]
```diff
@@ -33,6 +33,7 @@ class DateFilter extends AbstractContextAwareFilter
     const EXCLUDE_NULL = 'exclude_null';
     const INCLUDE_NULL_BEFORE = 'include_null_before';
     const INCLUDE_NULL_AFTER = 'include_null_after';
+    const ALWAYS_INCLUDE_NULL = 'always_include_null';
```
Comment (antograssiot) [inline src/Bridge/Doctrine/Orm/Filter/DateFilter.php]: `INCLUDE_NULL` only maybe to be more inline with the existing options name ?

File: src/Bridge/Doctrine/Orm/Filter/DateFilter.php [hunk 2]
```diff
@@ -190,7 +191,8 @@ protected function addWhere(QueryBuilder $queryBuilder, QueryNameGeneratorInterf
             $queryBuilder->andWhere($baseWhere);
         } elseif (
             (\in_array($operator, [self::PARAMETER_BEFORE, self::PARAMETER_STRICTLY_BEFORE], true) && self::INCLUDE_NULL_BEFORE === $nullManagement) ||
-            (\in_array($operator, [self::PARAMETER_AFTER, self::PARAMETER_STRICTLY_AFTER], true) && self::INCLUDE_NULL_AFTER === $nullManagement)
+            (\in_array($operator, [self::PARAMETER_AFTER, self::PARAMETER_STRICTLY_AFTER], true) && self::INCLUDE_NULL_AFTER === $nullManagement) ||
+            (\in_array($operator, [self::PARAMETER_AFTER, self::PARAMETER_STRICTLY_AFTER, self::PARAMETER_BEFORE, self::PARAMETER_STRICTLY_BEFORE], true) && self::ALWAYS_INCLUDE_NULL === $nullManagement)
```
Comment (antograssiot) [inline src/Bridge/Doctrine/Orm/Filter/DateFilter.php]: Maybe invert the conditions as the `===` might have a lower cost ?

**Comment by teohhanhui** [comment]
I'm not sure I understand the purpose of this. Aren't null values already included if you use one of `INCLUDE_NULL_BEFORE` or `INCLUDE_NULL_AFTER`?

**Comment by Simperfit** [comment]
Look at the linked issue:

The OP wants to include null values always and it seems that doesn't work for now, so it's like a merge of both after and before so it will always includes null values.

**Comment by teohhanhui** [comment]
Uhh, yeah I just looked at the issue... Maybe call it `INCLUDE_NULL_BEFORE_AND_AFTER`, to be explicit about what it actually does. I think it will cause confusion otherwise, when you somehow get null values at both ends.

**Comment by Simperfit** [comment]
done

**Comment by soyuka** [comment]
thanks @Simperfit

**Pull request merged**